src/XmlUtils.cc

```diff
@@ -27,17 +29,30 @@ inline namespace SDF_VERSION_NAMESPACE {
 /////////////////////////////////////////////////
 tinyxml2::XMLNode *DeepClone(tinyxml2::XMLDocument *_doc,
                               const tinyxml2::XMLNode *_src)
+{
+  sdf::Errors errors;
+  tinyxml2::XMLNode *result = DeepClone(_doc, _src, errors);
+  sdf::throwOrPrintErrors(errors);
+  return result;
+}
+
+/////////////////////////////////////////////////
+tinyxml2::XMLNode *DeepClone(tinyxml2::XMLDocument *_doc,
+                             const tinyxml2::XMLNode *_src,
+                             sdf::Errors &_errors)
 {
   if (_src == nullptr)
   {
-    sdferr << "Pointer to XML node _src is NULL\n";
+    _errors.push_back({sdf::ErrorCode::XML_ERROR,
```

Comment:
Can you add a test where `_src == nullptr`

Reply:
hmm, this PR is adding `sdf::Errors` objects. If it's trivial to add that test, okay, but if not, I think it might be out of scope.

Reply:
https://github.com/gazebosim/sdformat/pull/1296/commits/dd5f0d2a3d990406856f66f10371066016e23b69